torch/csrc/jit/python/init.cpp

```diff
@@ -294,47 +382,111 @@ class PythonSymFloatNodeImpl : public c10::SymFloatNodeImpl {
+    TORCH_CHECK(pother);
+    py::gil_scoped_acquire acquire;
+    auto r = getPyObj().attr(fname)(pother->getPyObj());
+    return c10::make_intrusive<PythonSymFloatNodeImpl>(r);
+  }
+
+  // (float, float) -> int
+  SymIntNode dispatch_common_int_(const char* fname, const SymFloatNode& other) {
+    auto pother = dynamic_cast<PythonSymFloatNodeImpl*>(other.get());
+    TORCH_CHECK(pother);
+    py::gil_scoped_acquire acquire;
+    auto r = getPyObj().attr(fname)(pother->getPyObj());
+    return c10::make_intrusive<PythonSymIntNodeImpl>(r);
+  }
+
+  // (float, int) -> int
+  SymIntNode dispatch_common_int_(const char* fname, const SymIntNode& other) {
+    auto pother = dynamic_cast<PythonSymIntNodeImpl*>(other.get());
+    TORCH_CHECK(pother);
+    py::gil_scoped_acquire acquire;
+    auto r = getPyObj().attr(fname)(pother->getPyObj());
+    return c10::make_intrusive<PythonSymIntNodeImpl>(r);
+  }
+
   SymFloatNode add(const SymFloatNode& other) override {
     return dispatch_common_(__FUNCTION__, other);
   }
+  SymFloatNode add(const SymIntNode& other) override {
+    return dispatch_common_(__FUNCTION__, other);
+  }
 
   SymFloatNode sub(const SymFloatNode& other) override {
     return dispatch_common_(__FUNCTION__, other);
   }
+  SymFloatNode sub(const SymIntNode& other) override {
+    return dispatch_common_(__FUNCTION__, other);
+  }
 
   SymFloatNode mul(const SymFloatNode& other) override {
     return dispatch_common_(__FUNCTION__, other);
   }
+  SymFloatNode mul(const SymIntNode& other) override {
+    return dispatch_common_(__FUNCTION__, other);
+  }
 
   SymFloatNode truediv(const SymFloatNode& other) override {
     return dispatch_common_(__FUNCTION__, other);
   }
+  SymFloatNode truediv(const SymIntNode& other) override {
+    return dispatch_common_(__FUNCTION__, other);
+  }
+
+  SymIntNode floordiv(const SymFloatNode& other) override {
+    return dispatch_common_int_(__FUNCTION__, other);
+  }
+  SymIntNode floordiv(const SymIntNode& other) override {
+    return dispatch_common_int_(__FUNCTION__, other);
+  }
 
   SymFloatNode pow(const SymFloatNode& other) override {
     return dispatch_common_(__FUNCTION__, other);
   }
+  SymFloatNode pow(const SymIntNode& other) override {
+    return dispatch_common_(__FUNCTION__, other);
+  }
 
   SymFloatNode eq(const SymFloatNode& other) override {
     return dispatch_common_(__FUNCTION__, other);
   }
+  SymFloatNode eq(const SymIntNode& other) override {
+    return dispatch_common_(__FUNCTION__, other);
+  }
 
   SymFloatNode gt(const SymFloatNode& other) override {
     return dispatch_common_(__FUNCTION__, other);
   }
+  SymFloatNode gt(const SymIntNode& other) override {
+    return dispatch_common_(__FUNCTION__, other);
+  }
 
   SymFloatNode lt(const SymFloatNode& other) override {
     return dispatch_common_(__FUNCTION__, other);
   }
+  SymFloatNode lt(const SymIntNode& other) override {
+    return dispatch_common_(__FUNCTION__, other);
```

Comment:
I ended up adding mixed overloads directly, which gave us twice as many functions (I tried to bundle them together though so they're easy to find).

I guess instead of that, I could have written all of the pybind functions to check for mixed types and upcast appropriately - so if anyone prefers that I can refactor. I confirmed that the current code passes all of the existing failing mixed types models (except for `hf_Reformer`, which blows up on `SymInt + FakeTensor`, which is an existing bug)